app/src/main/java/org/fossasia/openevent/utils/SpeakerIntent.java

```diff
@@ -0,0 +1,64 @@
+package org.fossasia.openevent.utils;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.view.View;
+import android.widget.ImageView;
+import android.widget.Toast;
+
+import org.fossasia.openevent.R;
+import org.fossasia.openevent.data.Speaker;
+
+/**
+ * Created by MananWason on 02-07-2015.
+ */
+public class SpeakerIntent {
+    Speaker speaker;
+    //    Intent intent;
+    Context context;
+
+    public SpeakerIntent(Speaker speaker, Context context) {
+        this.speaker = speaker;
+        this.context = context;
+    }
+
+    public void mIntent(final ImageView imageView) {
+        imageView.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                String url;
+                if (imageView.getTag().toString().equals(context.getString(R.string.linkedin))) {
```

Comment:
Why does this class need to know what the tag is? The class setting should specify whether it wants to set a Linkedin on click or github url as an on click parameter.